mindsdb/integrations/handlers/nuo_jdbc_handler/nuo_jdbc_handler.py

```diff
@@ -0,0 +1,291 @@
+                response = Response(
+                    RESPONSE_TYPE.ERROR,
+                    error_message=str(e)
+                )
+                self.connection.rollback()
+
+        if need_to_close is True:
+            self.disconnect()
+
+        return response
+
+    
+    def query(self, query: ASTNode) -> StatusResponse:
+        """Render and execute a SQL query.
+
+        Args:
+            query (ASTNode): The SQL query.
+
+        Returns:
+            Response: The query result.
+        """
+        if isinstance(query, ASTNode):
+            query_str = query.to_string()
+        else:
+            query_str = str(query)
+
+        return self.native_query(query_str)
+
+
+    def get_tables(self) -> StatusResponse:
+        """Get a list of all the tables in the database.
+
+        Returns:
+            Response: Names of the tables in the database.
+        """
+        if self.schema: 
+            query = f''' SELECT TABLENAME FROM SYSTEM.TABLES WHERE SCHEMA = '{self.schema}' '''
+        else: 
+            query = ''' SELECT TABLENAME FROM SYSTEM.TABLES WHERE SCHEMA != 'SYSTEM' '''
+    
+        result = self.native_query(query)
+        df = result.data_frame
+        result.data_frame = df.rename(columns={df.columns[0]: 'table_name'})
+        return result
+
+    
+    def get_columns(self, table_name: str) -> StatusResponse:
+        """Get details about a table.
+
+        Args:
+            table_name (str): Name of the table to retrieve details of.
+
+        Returns:
+            Response: Details of the table.
+        """
+
+        query = f''' SELECT FIELD FROM SYSTEM.FIELDS WHERE TABLENAME='{table_name}' '''
+        return self.native_query(query)
+    
+    
+connection_args = OrderedDict(
+    host={
+        'type': ARG_TYPE.STR,
+        'description': 'The host name or IP address of the NuoDB AP or TE. If is_direct is set to true then provide the TE IP else provide the AP IP.'
+    },
+    port={
+        'type': ARG_TYPE.INT,
+        'description': 'Specify port to connect to NuoDB. If is_direct is set to true then provide the TE port else provide the AP port.'
+    },
+    database={
+        'type': ARG_TYPE.STR,
+        'description': """
+            The database name to use when connecting with the NuoDB.
+        """
+    },
+    schema={
+        'type': ARG_TYPE.STR,
+        'description': """
+            The schema name to use when connecting with the NuoDB.
+        """
+    },
+    user={
+        'type': ARG_TYPE.STR,
+        'description': 'The username to authenticate with the NuoDB server.'
+    },
+    password={
+        'type': ARG_TYPE.STR,
+        'description': 'The password to authenticate the user with the NuoDB server.'
+    },
+    is_direct={
+        'type': ARG_TYPE.STR,
+        'description': 'This argument indicates whether a direct connection to the TE is to be attempted.'
+    },
+    jar_location={
```

Comment:
Should we add this to the README, looks important

Reply:
Done. added to README